test/api/AssetTest.ts

```diff
@@ -239,6 +249,107 @@ describe('Asset', function() {
+            testData.tenantContext.getTenant().subdomain,
+            testData.userContext
+          );
+        }
+        // Create a new Asset
+        testData.newAsset = await testData.userService.createEntity(
+          testData.userService.assetApi,
+          Factory.asset.build({
+            siteAreaID: testData.newSiteArea.id,
+            assetType: 'PR'
+          })
+        );
+        testData.createdAssets.push(testData.newAsset);
+        expect(testData.newAsset).to.not.be.null;
+        expect(config.get('assetConnectors.ioThink.url')).to.not.be.empty;
+        expect(config.get('assetConnectors.ioThink.user')).to.not.be.empty;
+        expect(config.get('assetConnectors.ioThink.password')).to.not.be.empty;
+        expect(config.get('assetConnectors.ioThink.meterID')).to.not.be.empty;
+      });
+
+      after(async () => {
+        await testData.userService.deleteEntity(
+          testData.userService.assetApi,
+          testData.newAsset
+        );
+      });
+
+      it('Should be able to create an ioThink asset connector', async () => {
+        const settings = await testData.centralUserService.settingApi.readAll({});
+        let newSetting = false;
+        let setting: SettingDB = settings.data.result.find((s) => s.identifier === TenantComponents.ASSET);
+        if (!setting) {
+          setting = {} as SettingDB;
+          setting.identifier = TenantComponents.ASSET;
+          newSetting = true;
+        }
+        const settingID = '0123456789abcdef';
+        setting.content = {
+          type: AssetSettingsType.ASSET,
+          asset: {
+            connections: [{
+              id: settingID,
+              name: 'Testing IoThink Asset Connector',
+              description: 'Testing IoThink Asset Connector',
+              url: config.get('assetConnectors.ioThink.url'),
+              timestamp: (new Date()),
+              type: AssetConnectionType.IOTHINK,
+              refreshIntervalMins: 1,
+              iothinkConnection: {
+                user: config.get('assetConnectors.ioThink.user'),
+                password: config.get('assetConnectors.ioThink.password')
+              }
+            }]
+          }
+        };
+        setting.sensitiveData = ['content.asset.connections[0].iothinkConnection.password'];
+        if (newSetting) {
+          await testData.centralUserService.settingApi.create(setting);
+        }
+        await testData.centralUserService.settingApi.update(setting);
+        const settingsUpdated = await testData.centralUserService.settingApi.readAll({});
+        const settingUpdated: SettingDB = settingsUpdated.data.result.find((s) => s.identifier === TenantComponents.ASSET);
+        expect(settingUpdated.content.asset.connections[0].id).to.be.eq(settingID);
+        testData.ioThinkAssetConnectorID = settingID;
+      });
+
+      it('should be able to connect ioThink asset connector connection', async () => {
```

Comment:
If the credentials are changed, the connection cannot be validated, as long as the token is still in the cache.

Reply:
I have added another test case that changes the settings of the connection and invalidates the connection. It seems to be working which should mean that the token is getting refreshed.